exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/FindPartitionConditions.java

```diff
@@ -217,8 +218,21 @@ public Void visitCorrelVariable(RexCorrelVariable correlVariable) {
   }
 
   public Void visitCall(RexCall call) {
-    // assume PUSH until proven otherwise
-    analyzeCall(call, PushDirFilter.PUSH);
+    boolean visited = false;
+    // examine the input of a CAST function; this could be extended for
+    // other functions in the future.
```

Comment:
Looks like DRILL-3560 reported issues for partition pruning, when it involves a general expression including UDF.  I think it would be fine that we address the general expression in DRILL-3560.